cmd/cli/plugin/managementcluster/featureflags.go

```diff
@@ -0,0 +1,21 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package main
+
+import (
+	"github.com/vmware-tanzu/tanzu-framework/tkg/constants"
+)
+
+// DefaultFeatureFlagsForManagementClusterPlugin is used to populate default feature-flags for the management-cluster plugin
+var (
+	DefaultFeatureFlagsForManagementClusterPlugin = map[string]bool{
+		"features.management-cluster.import":                       false,
+		"features.management-cluster.export-from-confirm":          true,
+		"features.management-cluster.standalone-cluster-mode":      false,
```

Comment:
I've been thinking further about these constants.  I think we have an opportunity here to tighten the CLIs API with its plugins.

I don't think it should be plugins that define the entire feature-flag path for their flags.  It gives them too much control over where flags will be stored.  They should only be able to specify a subpath under "features.\<pluginName\>", and it should be the CLI that prepends the fixed "features.\<pluginName\>" prefix.

For example, the feature-flag `features.management-cluster.standalone-cluster-mode` should only be `standalone-cluster-mode` in this file and passed that way to the CLI in the PluginDescriptor.  The CLI will then add the `features.management-cluster` prefix before storing it in the config file.

We can make this change without breaking backwards-compatibility, except for ""features.global.tkr-version-v1alpha3-beta" which is named wrong I believe.

We don't need to do this in this PR actually, but I think we should eventually do it.

What do you think?

What do you think?

Reply:
I agree we can do this improvement separately, but I also think we should do it before we separate the CLI code from TF as we could do it without dealing with compatibility.  I'll open an issue so we don't forget about it.

Reply:
I have opened #3824 to track this